Source/SwiftLintFramework/Rules/RuleConfigurations/DiscouragedDirectInitConfiguration.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  DiscouragedInitConfiguration.swift
+//  SwiftLint
+//
+//  Created by Ornithologist Coder on 8/1/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+private func toExplicitInitMethod(typeName: String) -> String {
+    return "\(typeName).init"
+}
+
+public struct DiscouragedDirectInitConfiguration: RuleConfiguration, Equatable {
+    public var severityConfiguration = SeverityConfiguration(.warning)
+
+    public var consoleDescription: String {
+        return severityConfiguration.consoleDescription + ", included: \(discouragedInits)"
+    }
+
+    public var severity: ViolationSeverity {
+        return severityConfiguration.severity
+    }
+
+    private(set) public var discouragedInits: Set<String>
+
+    private let defaultDiscouragedInits = [
+        "Bundle",
+        "UIDevice"
+    ]
+
+    init() {
+        discouragedInits = Set(defaultDiscouragedInits + defaultDiscouragedInits.map(toExplicitInitMethod))
+    }
+
+    // MARK: - RuleConfiguration
+
+    public mutating func apply(configuration: Any) throws {
+        guard let configuration = configuration as? [String: Any] else {
+            throw ConfigurationError.unknownConfiguration
+        }
+
+        if let severityString = configuration["severity"] as? String {
+            try severityConfiguration.apply(configuration: severityString)
+        }
+
+        if let included = [String].array(of: configuration["included"]) {
```

Comment:
actually, on a second thought, what do you think about using `"types"` as the key? I'm afraid that we already overload the term `included` too much.